### PR TITLE
Update TLS bootstrapping docs

### PR DESCRIPTION
**Pull request by lachie83**
Doc updates based on feedback in the following thread -- https://groups.google.com/forum/#!topic/kubernetes-sig-cluster-lifecycle/l0Dxlz_AWEY

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4490)

### CODE REVIEW

**Comment by lachie83** [comment]
cc @jbeda @mikedanese

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 1]
```diff
@@ -179,9 +186,10 @@ kubectl config set-credentials kubelet-bootstrap --token=${BOOTSTRAP_TOKEN} --ku
 
 When starting the kubelet, if the file specified by `--kubeconfig` does not exist, the bootstrap kubeconfig is used to request a client certificate from the API server. On approval of the certificate request and receipt back by the kubelet, a kubeconfig file referencing the generated key and obtained certificate is written to the path specified by `--kubeconfig`. The certificate and key file will be placed in the directory specified by `--cert-dir`.
 
-The flag to enable this bootstrapping when starting the kubelet is:
+The following flags are required to enable this bootstrapping when starting the kubelet is:
 
 ```
+--require-kubeconfig
 --experimental-bootstrap-kubeconfig="/path/to/bootstrap/kubeconfig"
```
Comment (ericchiang) [inline docs/admin/kubelet-tls-bootstrapping.md]: I think this is no longer experimental too. so the flag is now just `--bootstrap-kubeconfig`

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 2]
```diff
@@ -17,11 +17,18 @@ and progress on the feature is being tracked as [feature #43](https://github.com
...
 This group will later be used in the controller-manager configuration to scope approvals in the default approval
 controller. As this feature matures, you should ensure tokens are bound to a Role-Based Access Control (RBAC) policy which limits requests
 (using the bootstrap token) strictly to client requests related to certificate provisioning. With RBAC in place, scoping the tokens to a group allows for great flexibility (e.g. you could disable a particular bootstrap group's access when you are done provisioning the nodes).
 
+There are two methods by which tokens can be handled.
+
+1. [Bootstrap Tokens](/docs/admin/bootstrap-tokens.md)
+2. [Token authentication file](###token-authentication-file)
+
+Using bootstrap tokens is the preferred method as the token authentication file (csv) is difficult to manage especially in HA scenarios.
```
Comment (ericchiang) [inline docs/admin/kubelet-tls-bootstrapping.md]: Bootstrap tokens are alpha right now :/

Comment (lachie83) [inline docs/admin/kubelet-tls-bootstrapping.md]: Okay so I'll remove the preferred method wording and describe it as an alpha feature. How does that sound?

File: docs/admin/kubelet-tls-bootstrapping.md (hunk 1 above)
Comment (lachie83) [inline docs/admin/kubelet-tls-bootstrapping.md]: Okay. Fixing

File: docs/admin/kubelet-tls-bootstrapping.md (hunk 2 above)
Comment (ericchiang) [inline docs/admin/kubelet-tls-bootstrapping.md]: 👍 sounds great. I think they're slated to be beta in 1.8. So once that release comes around we can remove the warning.

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 3]
```diff
@@ -17,11 +17,18 @@ and progress on the feature is being tracked as [feature #43](https://github.com
 
 ## kube-apiserver configuration
 
-You must provide a token file which specifies at least one "bootstrap token" assigned to a kubelet bootstrap-specific group.
+You must provide a token which specifies at least one "bootstrap token" assigned to a kubelet bootstrap-specific group.
```
Comment (jbeda) [inline docs/admin/kubelet-tls-bootstrapping.md]: This is awkward.  How about:
```
You must configure a token based authentication mechanism that will provide authorization for a user in the `system:bootstrappers` group.
```

Note that this group is hard coded as part of the bootstrap token mechanism but is configurable with the (should be deprecated) token auth file.  We should standardize the rest of the instructions in this file to that group.

Specifically, change references to `system:kubelet-bootstrap` or `kubelet-bootstrap-token` to `system:bootstrappers` everywhere below.  (Right now, even, the instructions are wrong as the CSV example uses one group but the RBAC rules use another.)

Comment (lachie83) [inline docs/admin/kubelet-tls-bootstrapping.md]: Appreciate the feedback. I tripped on that too. Will update

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 4]
```diff
@@ -179,10 +187,11 @@ kubectl config set-credentials kubelet-bootstrap --token=${BOOTSTRAP_TOKEN} --ku
 
 When starting the kubelet, if the file specified by `--kubeconfig` does not exist, the bootstrap kubeconfig is used to request a client certificate from the API server. On approval of the certificate request and receipt back by the kubelet, a kubeconfig file referencing the generated key and obtained certificate is written to the path specified by `--kubeconfig`. The certificate and key file will be placed in the directory specified by `--cert-dir`.
 
-The flag to enable this bootstrapping when starting the kubelet is:
+The following flags are required to enable this bootstrapping when starting the kubelet is:
```
Comment (kris-nova) [inline docs/admin/kubelet-tls-bootstrapping.md]: So this is the big win IMHO - should we add a large "notice" or "note" heading here?

Comment (lachie83) [inline docs/admin/kubelet-tls-bootstrapping.md]: I'd be happy to. @ericchiang @jbeda -- Thoughts?

Comment (ericchiang) [inline docs/admin/kubelet-tls-bootstrapping.md]: "Note:" is what the kubeadm docs do https://github.com/kubernetes/kubernetes.github.io/blob/master/docs/setup/independent/install-kubeadm.md#installing-kubelet-and-kubeadm

"Note: The following flags..." sgtm

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 5]
```diff
@@ -17,11 +17,19 @@ and progress on the feature is being tracked as [feature #43](https://github.com
 
 ## kube-apiserver configuration
 
-You must provide a token file which specifies at least one "bootstrap token" assigned to a kubelet bootstrap-specific group.
+You must configure a token based authentication mechanism that will provide authorization for a user in the `system:bootstrappers` group.
```
Comment (ericchiang) [inline docs/admin/kubelet-tls-bootstrapping.md]: Authentication mechanisms don't provide authorization.

Maybe:

> The API server should be configured with an [authenticator](/docs/admin/authentication/) that can authenticate tokens as a user in the `system:bootstrappers` group.

**Comment by lachie83** [comment]
@jbeda Can you kindly confirm that the group is indeed system:bootstrappers and NOT system:node-bootstrapper (ref https://kubernetes.io/docs/admin/authorization/rbac/ Other Component Roles)

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 6]
```diff
@@ -17,11 +17,19 @@ and progress on the feature is being tracked as [feature #43](https://github.com
...
+
 This group will later be used in the controller-manager configuration to scope approvals in the default approval
 controller. As this feature matures, you should ensure tokens are bound to a Role-Based Access Control (RBAC) policy which limits requests
 (using the bootstrap token) strictly to client requests related to certificate provisioning. With RBAC in place, scoping the tokens to a group allows for great flexibility (e.g. you could disable a particular bootstrap group's access when you are done provisioning the nodes).
 
+There are two methods by which tokens can be handled.
```
Comment (mikedanese) [inline docs/admin/kubelet-tls-bootstrapping.md]: There are way more than two. Is this really in scope for this doc?

**Comment by ericchiang** [comment]
>  Can you kindly confirm that the group is indeed system:bootstrappers and NOT system:node-bootstrapper

ag says 'system:bootstrappers':

```
$ ag 'system:bootstrappers'
plugin/pkg/auth/authenticator/token/bootstrap/bootstrap.go
65:// "system:bootstrappers" group and with the "system:bootstrap:(token-id)" username.

plugin/pkg/auth/authenticator/token/bootstrap/bootstrap_test.go
84:				Groups: []string{"system:bootstrappers"},
197:				Groups: []string{"system:bootstrappers"},

pkg/bootstrap/api/types.go
66:	// user "system:bootstrap:<token-id>" in the group "system:bootstrappers".
84:	BootstrapGroup = "system:bootstrappers"
$ ag 'system:node-bootstrapper '
$
```

File: docs/admin/kubelet-tls-bootstrapping.md (hunk 6 above)
Comment (lachie83) [inline docs/admin/kubelet-tls-bootstrapping.md]: How should we scope this down. Can all methods be used in the context of TLS bootstrapping nodes?

Comment (ericchiang) [inline docs/admin/kubelet-tls-bootstrapping.md]: Any auth-N strategy can be used for TLS bootstrapping. You could use an x509 cert or service account token as the initial credentials for the kubelet, the token file is just the way we've been suggesting this.

Maybe

> While any authentication strategy can be used for the kubelet's initial bootstrap credentials, the following two authenticators are recommended for ease of provisioning.

Comment (mikedanese) [inline docs/admin/kubelet-tls-bootstrapping.md]: Eric's suggestion works for me. Thanks.

**Comment by lachie83** [comment]
Thanks @mikedanese @ericchiang I've updated based on your review

**Comment by chenopis** [comment]
Deploy preview ready!

Built with commit 6b79c307ae1f79cf5411daba18013d451ea227cb

https://deploy-preview-4490--kubernetes-io-master-staging.netlify.com

**Comment by jbeda** [comment]
/lgtm

Sorry for dropping the balls.  GitHub notification noise.  In future feel free to ping me on slack.

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 7]
```diff
@@ -17,11 +17,19 @@ and progress on the feature is being tracked as [feature #43](https://github.com
 
 ## kube-apiserver configuration
 
-You must provide a token file which specifies at least one "bootstrap token" assigned to a kubelet bootstrap-specific group.
+The API server should be configured with an [authenticator](/docs/admin/authentication.md) that can authenticate tokens as a user in the `system:bootstrappers` group.
```
Comment (chenopis) [inline docs/admin/kubelet-tls-bootstrapping.md]: We don't use the .md extension in links, so the authenticator URL should be: `/docs/admin/authentication/`

File: docs/admin/kubelet-tls-bootstrapping.md [hunk 8]
```diff
@@ -17,11 +17,19 @@ and progress on the feature is being tracked as [feature #43](https://github.com
...
+
 This group will later be used in the controller-manager configuration to scope approvals in the default approval
 controller. As this feature matures, you should ensure tokens are bound to a Role-Based Access Control (RBAC) policy which limits requests
 (using the bootstrap token) strictly to client requests related to certificate provisioning. With RBAC in place, scoping the tokens to a group allows for great flexibility (e.g. you could disable a particular bootstrap group's access when you are done provisioning the nodes).
 
+While any authentication strategy can be used for the kubelet's initial bootstrap credentials, the following two authenticators are recommended for ease of provisioning.
+
+1. [Bootstrap Tokens](/docs/admin/bootstrap-tokens.md) - __alpha__
```
Comment (chenopis) [inline docs/admin/kubelet-tls-bootstrapping.md]: similar issue to above, URL should be: `/docs/admin/bootstrap-tokens/`

**Comment by lachie83** [comment]
@chenopis Thanks for the review. I've updated to links as requested.
